apps/chatbots/views.py

```diff
@@ -517,6 +517,21 @@ def chatbot_session_details_view(request, team_slug: str, experiment_id: uuid.UU
     )
 
 
+@require_POST
+@login_and_team_required
+@permission_required("experiments.change_experimentsession", raise_exception=True)
+def end_chatbot_session(request, team_slug: str, experiment_id: uuid.UUID, session_id: str):
+    experiment_session = get_object_or_404(
+        ExperimentSession,
+        experiment__public_id=experiment_id,
+        external_id=session_id,
+        team=request.team,
+    )
+    experiment_session.end(propagate=request.POST.get("fire_end_event", "true") == "true")
```

Comment:
nit: extract var for easier reading
```suggestion
    propagate_event = request.POST.get("fire_end_event", "true") == "true"
    experiment_session.end(propagate=propagate_event)
```